extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -186,6 +205,33 @@ private List<Throwable> validateInterceptorBindingTarget(AnnotationInstance bind
         return throwables;
     }
 
+    private Optional<DotName> findCacheKeyGenerator(AnnotationInstance binding, AnnotationTarget target) {
+        if (target.kind() == METHOD && (CACHE_RESULT.equals(binding.name()) || CACHE_INVALIDATE.equals(binding.name()))) {
+            AnnotationValue keyGenerator = binding.value("keyGenerator");
+            if (keyGenerator != null) {
+                return Optional.of(keyGenerator.asClass().name());
+            }
+        }
+        return Optional.empty();
+    }
+
+    // Key generators must have a default constructor if they are not managed by Arc.
+    private List<Throwable> validateKeyGeneratorsDefaultConstructor(CombinedIndexBuildItem combinedIndex,
+            BeanDiscoveryFinishedBuildItem beanDiscoveryFinished, Set<DotName> keyGenerators) {
+        List<DotName> managedBeans = beanDiscoveryFinished.geBeans()
```

Comment:
If only managed beans are supported then you should filter out unsupported bean types, i.e. `beanDiscoveryFinished.geBeans().stream().filter(BeanInfo::isClassBean)`.